theme/theme.go

```diff
@@ -11,28 +11,105 @@ import (
 
 const (
 	// ColorRed is the red primary color name
+	//
+	// Since: 1.4.0
 	ColorRed = "red"
 	// ColorOrange is the orange primary color name
+	//
+	// Since: 1.4.0
 	ColorOrange = "orange"
 	// ColorYellow is the yellow primary color name
+	//
+	// Since: 1.4.0
 	ColorYellow = "yellow"
 	// ColorGreen is the green primary color name
+	//
+	// Since: 1.4.0
 	ColorGreen = "green"
 	// ColorBlue is the blue primary color name
+	//
+	// Since: 1.4.0
 	ColorBlue = "blue"
 	// ColorPurple is the purple primary color name
+	//
+	// Since: 1.4.0
 	ColorPurple = "purple"
 	// ColorBrown is the brown primary color name
+	//
+	// Since: 1.4.0
 	ColorBrown = "brown"
 	// ColorGray is the gray primary color name
+	//
+	// Since: 1.4.0
 	ColorGray = "gray"
 )
 
+var (
+	// Colors specifies each of the known colour names that a theme can contain.
+	//
+	// Since 2.0.0
+	Colors = struct {
+		Background     fyne.ThemeColorName
+		Text           fyne.ThemeColorName
+		Button         fyne.ThemeColorName
+		DisabledButton fyne.ThemeColorName
+		DisabledText   fyne.ThemeColorName
+		Focus          fyne.ThemeColorName
+		Hover          fyne.ThemeColorName
+		PlaceHolder    fyne.ThemeColorName
+		Primary        fyne.ThemeColorName
+		ScrollBar      fyne.ThemeColorName
+		Shadow         fyne.ThemeColorName
+	}{
+		"background",
+		"text",
+		"button",
+		"disabledButton",
+		"disabledText",
+		"focus",
+		"hover",
+		"placeholder",
+		"primary",
+		"scrollBar",
+		"shadow",
+	}
+
+	// Sizes specifies each of the known size names that a theme can contain.
+	//
+	// Since 2.0.0
+	Sizes = struct {
+		Padding        fyne.ThemeSizeName
+		ScrollBar      fyne.ThemeSizeName
+		ScrollBarSmall fyne.ThemeSizeName
+		Text           fyne.ThemeSizeName
+		InlineIcon     fyne.ThemeSizeName
+	}{
+		"paddinng",
+		"scrollBar",
+		"scrollBarSmall",
+		"text",
+		"iconInline",
+	}
+
+	// Variants lists each of the known theme variants that can apply to a color lookup.
+	//
+	// Since 2.0.0
+	Variants = struct {
+		Dark  fyne.ThemeVariant
+		Light fyne.ThemeVariant
+		// potential for adding theme types such as high visibility or monochrome...
+	}{
+		1,
```

Comment:
If this starts at 1, what does 0 represent?

Reply:
Good catch, there was an "unspecified" concept for a while but I removed it - revising now